lib/spack/docs/packaging_guide.rst

```diff
@@ -1423,6 +1423,37 @@ other similar operations:
             ).with_default('auto').with_non_feature_values('auto'),
         )
 
+"""""""""""""""""""""""""""
+Conditional Possible Values
+"""""""""""""""""""""""""""
+
+There are cases where a variant may take multiple values, and the list of allowed values
+expand over time. Think for instance at the C++ standard with which we might compile
+Boost, which can take one of multiple possible values with the latest standards
+only available from a certain version on.
+
+To model a similar situation we can use *conditional possible values* in the variant declaration:
+
+.. code-block:: python
+
+   variant(
+       'cxxstd', default='98',
+       values=(
+           '98', '11', '14',
+           # C++17 is not supported by Boost < 1.63.0.
+           value('17', when='@1.63.0:'),
```

Comment:
So that nobody starts asking for this:

```python
when(
    value('17'),
    value('18'),
    value('19'),
    value('20'),
    when='@1.63.0:',
)
```

we should probably allow this:

```python
    value('17', '18', '19', '20', when='@1.63.0:')
```

Reply:
I'll try to check what is needed for this modification. Can we use another name though instead of `value` if we have to handle lists?

Reply:
I agree, `value` is singular, it's looks odd when it takes multiple `values`.

Reply:
`values` / `value_list`?

Reply:
What about:
```python
variant(
    'cxxstd', default='98',
    values=(
        '98', '11', '14',
        # C++17 is not supported by Boost < 1.63.0.
        conditional('17', when='@1.63.0:'),
        # C++20/2a is not support by Boost < 1.73.0
        conditional('2a', '2b', when='@1.73.0:')
    )
)
```
?